daffodil-lib/src/main/scala/org/apache/daffodil/schema/annotation/props/ByHandMixins.scala

```diff
@@ -432,7 +432,9 @@ sealed trait EmptyElementParsePolicy extends EmptyElementParsePolicy.Value
 object EmptyElementParsePolicy extends Enum[EmptyElementParsePolicy] {
   case object TreatAsMissing extends EmptyElementParsePolicy
   case object TreatAsEmpty extends EmptyElementParsePolicy
-  override lazy val values = Array(TreatAsMissing, TreatAsEmpty)
+  case object TreatAsAbsent extends EmptyElementParsePolicy
+
+  override lazy val values = Array(TreatAsMissing, TreatAsEmpty, TreatAsAbsent)
 
   def apply(name: String, context: ThrowsSDE): EmptyElementParsePolicy = stringToEnum("emptyElementParsePolicy", name, context)
```

Comment:
Around line 470 in this file, we have this comment:
```scala
// This property is an extension, so we don't want to require users to
// add this property just to silence this warning, especially since the
// property might change in the future. So silently use a default without
// outputting a warning. We may want to turn this warning on when the
// property makes its way into the official DFDL spec.
//SDW(
//  WarnID.EmptyElementParsePolicyError,
//  "Property 'dfdlx:emptyElementParsePolicy' is required but not defined, using tunable '%s' by default.",
//  defaultEmptyElementParsePolicy)
```
Do we want to re-enable this warning on now that this property is in the spec?

Do we also want to remove the defaulting behavior? I imagine no, since it might break a number of schemas? I guess after having this warning on for some number of releases we can flip the switch and require this property.

Do we want to add a value to DFDLGeneralFormat.dfdl.xsd? Is there an obvious choice or is it really dependent on the file format?

Reply:
@mbeckerle thoughts on the above questions?

Reply:
We should re-enable this warning, yes. Making the property required right now would break people's schemas. 

We do need to also add this property to DFDLGeneralFormat,  with value treatAsEmpty. But this may need to be in Daffodil-only part of that if IBM DFDL doesn't tolerate the property.

Reply:
> we want to add the property to DFDLGeneralFormat.dfdl.xsd and give it the same 'treatAsEmpty' default value. 

I just thought of one potential issue when adding to DFDLGeneralFormat. If IBM DFDL doesn't support this new property then any schemas that used DFDLGeneralFormat.dfdl.xsd might now error when cross run with IBM? To ensure backwards compat, maybe we don't want to add the property to the file, especially since it's optional (as Mike says) and we already have built in defaulting behavior? Though as you point out, that would mean lots of schemas would need to manually add this property to suppress the warning.

Reply:
IBM should have its own DFDLGeneralFormat.dfdl.xsd so that schemas can import DFDLGeneralFormat.dfdl.xsd without worrying about problems like that, yes?  If the IBM cross tester is using Daffodil's DFDLGeneralFormat.dfdl.xsd with IBM's DFDL processor, that seems like a behavior which needs to be changed, don't you think?

Reply:
DFDLGeneralFormat.dfdl.xsd is a thing Daffodil invented, I don't think IBM has any concept of it. DFDL schemas use the full path of `org/apache/daffodil/xsd/DFDLGeneralFormat.dfdl.xsd `, which is daffodil specific and what the cross tester will use since that is what schemas import.

I guess we do have two different defineFormats: `DFDLGeneralFormat` and `DFDLGeneralFormatPortable`. We could put the new property in the DFDLGeneralFormat define, but many schemas (including most of our tests) don't use the "portable" variant and will probably fail if cross tested. Maybe that's okay?

Reply:
Yes, tests that are portable have to use GeneralFormatPortable because there are some other subtle incompatibilities between IBM DFDL and Daffodil - things about time zones (like Z vs. -00:00 suffixes equivalent values, but not identical, so the same test doesn't compare correctly), lack of support for encodingErrorPolicy 'replace', etc.

Reply:
To clarify, both GeneralFormat and GeneralFormatPortable are defined in DFDLGeneralFormat.dfdl.xsd. They share a base format with all the fully portable properties.

Reply:
I've added the property to `DFDLGeneralFormat.dfdl.xsd` and re-enabled the warning. This required an update to a test where it was relying on the property to be missing and defaulting. I've updated that test to use a different schema instead of using the tuneable.